tests/rbd/test_rbd_deep_flatten_negative.py

```diff
@@ -0,0 +1,111 @@
+                if out or err:
+                    log.error(f"Feature deep-flatten was not disabled for {image}")
+                    return 1
+                info_spec = {"image-or-snap-spec": f"{pool}/{image}", "format": "json"}
+                out, err = rbd.info(**info_spec)
+                if err:
+                    log.info(f"Error while fetching info for image {image}")
+                    return 1
+                log.info(f"Image info: {out}")
+
+                out_json = json.loads(out)
+                if "deep-flatten" in out_json["features"]:
+                    log.error(f"Feature deep-flatten was not disabled for {image}")
+                    return 1
+
+                snap_spec = f"{pool}/{image}@snap_{image}"
+                clone_spec = f"{pool}/clone_{image}"
+                create_snap_and_clone(rbd, snap_spec, clone_spec)
+                # Create snapshot for the clone
+                clone_snap_spec = {
+                    "snap-spec": f"{pool}/clone_{image}@snap_clone_{image}"
+                }
+                out, err = rbd.snap.create(**clone_snap_spec)
+                if out or err and "100% complete" not in err:
+                    log.error(f"Snapshot creation failed for {clone_snap_spec}")
+                    return 1
+
+                flatten_config = {"image-spec": clone_spec}
+                out, err = rbd.flatten(**flatten_config)
+                if out or err and "100% complete" not in err:
+                    log.error(f"Flatten clone failed for {clone_spec}")
+                    return 1
+
+                snap_conf = {"snap-spec": snap_spec}
+                out, err = rbd.snap.unprotect(**snap_conf)
+                if out or err:
+                    err_msg = f"Snapshot unprotect failed as expected for {snap_spec} since deep-flatten"
+                    err_msg += f" is not enabled on the image {image} and clone contains snapshots"
+                    log.info(err_msg)
+                    log.info(f"Error message for snapshot unprotect: {out} {err}")
+                else:
+                    log.error(
+                        "Snapshot unprotect succeeded even though clone had snapshots"
+                    )
+                    return 1
+    return 0
+
+
+def run(**kw):
+    """Test parent snapshot deletion on an image where deep-flatten is disabled
+     (Negative scenario)
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common and fio packages,
+    conf and keyring files
+
+    Test cases covered -
+    1) CEPH-9831 - Test parent snapshot deletion on an image where deep-flatten
+     is disabled (Negative scenario)
+
+    Test Case Flow
+    1. Create image and disable deep-flatten feature on that image
+    2. Create a snapshot on this image, protect the snapshot and create a clone on that image
+    3. Create a snapshot for the clone
+    4. Flatten the clone
+    5. Unprotect the parent snapshot
+
+    """
+    pool_types = []
```

Comment:
if pool_types is empty how it will run for EC and REP pool.

Reply:
See Line 101, pool_types is getting updated there.